File: cloud/oauth-server/src/main/java/pers/acp/admin/oauth/token/store/SecurityTokenStoreUtil.java
package pers.acp.admin.oauth.token.store;

import pers.acp.admin.oauth.token.LoginLog;
import pers.acp.admin.oauth.vo.LoginLogVO;

import java.util.List;

/**
 * @author zhang by 08/03/2019
 * @since JDK 11
 */
class SecurityTokenStoreUtil {

    static void parseLoginLogVO(List<LoginLogVO> loginLogVOList, LoginLog loginLog) {
        boolean process = false;
        for (LoginLogVO loginLogVO : loginLogVOList) {
            if (loginLogVO.getAppid().equals(loginLog.getAppid()) && loginLogVO.getDate().equals(loginLog.getDate())) {
                loginLogVO.setCount(loginLogVO.getCount() + 1);
                process = true;
                break;
            }
        }
        if (!process) {
            LoginLogVO loginLogVO = new LoginLogVO();
            loginLogVO.setAppid(loginLog.getAppid());
            loginLogVO.setDate(loginLog.getDate());
            loginLogVO.setCount(1);
            loginLogVOList.add(loginLogVO);
        }
    }

}

File: cloud/oauth-server/src/main/kotlin/pers/acp/admin/oauth/token/store/SecurityTokenStoreUtil.kt
package pers.acp.admin.oauth.token.store

import pers.acp.admin.oauth.token.LoginLog
import pers.acp.admin.oauth.vo.LoginLogVo

/**
 * @author zhang by 08/03/2019
 * @since JDK 11
 */
internal object SecurityTokenStoreUtil {

    fun parseLoginLogVO(loginLogVoList: MutableList<LoginLogVo>, loginLog: LoginLog) {
        var process = false
        for (loginLogVO in loginLogVoList) {
            if (loginLogVO.appId == loginLog.appId && loginLogVO.date == loginLog.date) {
                loginLogVO.count = loginLogVO.count + 1
                process = true
                break
            }
        }
        if (!process) {
            loginLogVoList.add(LoginLogVo(
                    appId = loginLog.appId,
                    date = loginLog.date,
                    count = 1
            ))
        }
    }

}
